scoring/src/main/scala/ai/h2o/sparkling/ml/models/H2OMOJOPipelineModel.scala

```diff
@@ -118,17 +133,35 @@ class H2OMOJOPipelineModel(override val uid: String)
           }
       }
 
+      def mojoFrameToArray(mf : MojoFrame) = {
+        mf.getColumnNames.zipWithIndex.map {
+          case (_, i) =>
+            val columnData = mf.getColumnData(i).asInstanceOf[Array[_]]
+            if (columnData.length != 1) {
+              throw new RuntimeException("Invalid state, we predict on each row by row, independently at this moment.")
+            }
+            columnData(0)
+        }
+      }
+
       builder.addRow(rowBuilder)
-      val output = mojoPipeline.transform(builder.toMojoFrame)
-      val predictions = output.getColumnNames.zipWithIndex.map {
-        case (_, i) =>
-          val columnData = output.getColumnData(i).asInstanceOf[Array[_]]
-          if (columnData.length != 1) {
-            throw new RuntimeException("Invalid state, we predict on each row by row, independently at this moment.")
-          }
-          columnData(0)
+      val inputMojoFrame = builder.toMojoFrame
+
+      val outputPredictions = mojoPipeline.transform(inputMojoFrame)
+      val predictions = mojoFrameToArray(outputPredictions)
+
+      val contributions = if (getWithContributions()) {
+        val outputContributions = mojoPipelineContributions.transform(inputMojoFrame)
+        mojoFrameToArray(outputContributions)
+      } else {
+        Array[Any]()
+      }
+
+      val content = if (getNamedMojoOutputColumns()) {
+        predictions ++ contributions
+      } else {
+        Array[Any](predictions ++ contributions)
```

Comment:
I don't understand that part - returning `predictions ++ contributions` or returning it wrapped in an array

Reply:
Depending on the getNamedMojoOutputColumns() value, the content has to have slightly different structure to match a schema which is also dependent on the getNamedMojoOutputColumns(). Here, either Array(val1, val2, ...) or Array(Array(val1, val2, ...)) gets returned.